app/src/main/java/com/aconno/sensorics/ui/MainActivity2.kt

```diff
@@ -344,15 +339,25 @@ class MainActivity2 : DaggerAppCompatActivity(),
         }
     }
 
+    inner class PageChangedCallback : ViewPager2.OnPageChangeCallback() {
```

Comment:
I think it would be a good idea to move these inner classes to their own files, in order to separate the responsibilities, what do you think?

Reply:
We need 'viewPagerAdapter' instance to make  'PageChangedCallback' and also we need to call  invalidateOptionsMenu(). Thats why PageChangedCallback should stay in here. Otherwise we need to pass activity instance etc.

But ViewPagerAdapter can be seperate file. I think.